File: AndroidStudio_folder/Message2021/app/src/main/java/com/example/message2021/MainActivity.java
package com.example.message2021;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    //전송버튼을 클릭하면 호출된다.
    public void onSendMessage(View view){
        EditText et_msg = findViewById(R.id.et_msg);
        String msg = et_msg.getText().toString();
        Intent intent = new Intent(this, ReceiveActivity.class);

        intent.putExtra(ReceiveActivity.EXTRA_MESSAGE,msg);
        startActivity(intent);
    }

}

File: AndroidStudio_folder/KT_FCMYSC801/app/src/main/java/com/example/kt_fcmysc801/MainActivity.kt
package com.example.kt_fcmysc801

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.google.firebase.messaging.FirebaseMessaging

class MainActivity : AppCompatActivity() {
    //게으른 인스턴스화
    private val resultTxtView: TextView by lazy {
        findViewById(R.id.resultTextView)
    }

    private val firebaseTokenText: TextView by lazy {
        findViewById(R.id.firebaseTokenText)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initFirebase();
    }

    private fun initFirebase(){
        FirebaseMessaging.getInstance().token
            .addOnCompleteListener { task ->
                if(task.isSuccessful){
                    firebaseTokenText.text = task.result
                }
            }
    }
}